quickfixj-core/src/main/java/quickfix/Session.java

```diff
@@ -2605,19 +2616,12 @@ private boolean sendRaw(Message message, int num) {
                     logApplicationException("toApp()", t);
                 }
                 messageString = message.toString();
+                persist(message, num);
```

Comment:
Could you please pass the `messageString` into `persist()`. That avoids two calls to `toString()` which should be avoided since it recalculates some fields of the message.